xrpl/asyncio/transaction/reliable_submission.py

```diff
@@ -34,8 +35,18 @@ async def _wait_for_final_transaction_outcome(
     # new persisted transaction
 
     # query transaction by hash
-    transaction_response = await client._request_impl(Tx(transaction=transaction_hash))
-
+    try:
+        transaction_response = await client._request_impl(
+            Tx(transaction=transaction_hash)
+        )
+    except XRPLRequestFailureException as e:
+        if e.error == "txnNotFound":
+            # err code for if the txn is not found on the ledger due to race condition
+            return await _wait_for_final_transaction_outcome(
+                transaction_hash, client, prelim_result
+            )
```

Comment:
Can this result in infinite calls? What if the error message doesn't change and remains `txnNotFound`?

Reply:
The `sleep` restricts it to one call per second. It _should_ eventually change, but I guess it's hard to test the "stuck in the queue" scenario (and I don't know if that shows up on other nodes, but I _think_ it does).

Reply:
this could - we can add a check to stop after couple calls and return error

Reply:
I don't think the JS solutions covers the scenario

Reply:
Yeah, I think it's best to have a maximum attempts check and error out if the status doesn't change.

Reply:
might have to add an additional arg to `_wait_for_final_transaction_outcome` to keep track of attempts, how many attempts do you think we should allow?

Reply:
3 or 4 sounds like a good number if restricting it to one call per second.